test/httpoison_base_test.exs

```diff
@@ -105,4 +105,17 @@ defmodule HTTPoisonBaseTest do
 
     assert validate :hackney
   end
+
+  test "passing proxy option with proxy_auth" do
+    expect(:hackney, :request, [{[:post, "http://localhost", [], "body", [proxy: "proxy", proxy_auth: {"username", "password"}]],
```

Comment:
You just need to swap the order of `proxy` and `proxy_auth`. This is related to the order we prepend the options list. 

 `expect(:hackney, :request, [{[:post, "http://localhost", [], "body", [proxy_auth: {"username", "password"}, proxy: "proxy"]],, {:ok, 200, "headers", :client}}])`